### DISCUSSION TITLE
[mesheryctl] New subcommand: `system dashboard`

### DISCUSSION THREAD

**Discussion by leecalcote** [meshery/meshery#2808] [Ideas]
### Current Situation
Meshery UI's URL is presented to users on `system start`, but it not recallable by a subsequent command after initial start.

### Proposal for Discussion
Have a `mesheryctl system dashboard` that invokes this logic:

1. test connection to Meshery UI
   1. if available, open the user's default browser
   1. if not available, and current-context's `platform` is "kubernetes", then initiate a kube proxy tunnel.
      1. suggest that users run this command with `&` in order to background the tunnel

### Benefits to Users
1. Ability to recall the location of their Meshery UI
2. Ability to bypass what might be challenging K8s networking by providing a tunnel from local machine to K8s deployment of Meshery.

### Drawbacks
1. If Meshery UI is ever deployed as a component separate from Meshery Server, this command's behavior will need to be evaluated.
1. Benefit #2 is mostly only useful on K8s. _<-- is that true?_

### Prior Art (other examples)
- `backyards dashboard` - https://banzaicloud.com/docs/backyards/cli/reference/backyards_dashboard/
- `linkerd dashboard`

**Comment by pottekkat** [comment]
This could be added to mesheryctl with the recent works by @metonymic-smokey to get the endpoint to which Meshery is deployed and opening it on the browser automatically. So, I guess it should open up the UI even for the Kubernetes platform. If not then port-forward?

**Reply by leecalcote** [reply]
This sounds about right to me, @navendu-pottekkat, yes.

**Reply by pottekkat** [reply]
Okay, I will open up an issue for this

**Reply by piyushsingariya** [reply]
@navendu-pottekkat Issue hasn't been opened up yet!

**Reply by metonymic-smokey** [reply]
@navendu-pottekkat I think the endpoint returned for kubernetes by `GetServiceEndpoint` is still incorrect?  
Do you think it would be better to wait till that gets sorted?

**Reply by pottekkat** [reply]
We are also tunnelling to the k8s cluster right? So the endpoint could be updated?

**Reply by pottekkat** [reply]
I will open up this issue and we can discuss on the specifics
